premake5.lua

```
@@ -0,0 +1,139 @@
+    cdialect "C11"
+    startproject "example"
+
+    -- Detect architecture automatically
+    filter "architecture:x86_64 or architecture:x64"
+        architecture "x86_64"
+    filter "architecture:arm64 or architecture:aarch64"
+        architecture "ARM64"
+    filter {}
+
+    -- Platform-specific settings
+    filter "system:macosx"
```

Luacheck (1.2.0)

[warning] accessing undefined variable 'filter'

(W113)